samples/bluetooth/central_peripheral_hr/src/main.c

```diff
@@ -0,0 +1,458 @@
+
+	printk("[NOTIFICATION] data %p length %u\n", data, length);
+
+	total_rx_count++;
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static uint8_t discover_func(struct bt_conn *conn,
+			     const struct bt_gatt_attr *attr,
+			     struct bt_gatt_discover_params *params)
+{
+	int err;
+
+	if (!attr) {
+		printk("Discover complete\n");
+		(void)memset(params, 0, sizeof(*params));
+		return BT_GATT_ITER_STOP;
+	}
+
+	printk("[ATTRIBUTE] handle %u\n", attr->handle);
+
+	if (!bt_uuid_cmp(discover_params.uuid, BT_UUID_HRS)) {
+		memcpy(&discover_uuid, BT_UUID_HRS_MEASUREMENT, sizeof(discover_uuid));
+		discover_params.uuid = &discover_uuid.uuid;
+		discover_params.start_handle = attr->handle + 1;
+		discover_params.type = BT_GATT_DISCOVER_CHARACTERISTIC;
+
+		err = bt_gatt_discover(conn, &discover_params);
+		if (err) {
+			printk("Discover failed (err %d)\n", err);
+		}
+	} else if (!bt_uuid_cmp(discover_params.uuid,
+				BT_UUID_HRS_MEASUREMENT)) {
+		memcpy(&discover_uuid, BT_UUID_GATT_CCC, sizeof(discover_uuid));
+		discover_params.uuid = &discover_uuid.uuid;
+		discover_params.start_handle = attr->handle + 2;
+		discover_params.type = BT_GATT_DISCOVER_DESCRIPTOR;
+		subscribe_params.value_handle = bt_gatt_attr_value_handle(attr);
+
+		err = bt_gatt_discover(conn, &discover_params);
+		if (err) {
+			printk("Discover failed (err %d)\n", err);
+		}
+	} else {
+		subscribe_params.notify = notify_func;
+		subscribe_params.value = BT_GATT_CCC_NOTIFY;
+		subscribe_params.ccc_handle = attr->handle;
+
+		err = bt_gatt_subscribe(conn, &subscribe_params);
+		if (err && err != -EALREADY) {
+			printk("Subscribe failed (err %d)\n", err);
+		} else {
+			printk("[SUBSCRIBED]\n");
+		}
+
+		return BT_GATT_ITER_STOP;
+	}
+
+	return BT_GATT_ITER_STOP;
+}
+
+static bool eir_found(struct bt_data *data, void *user_data)
+{
+	bt_addr_le_t *addr = user_data;
+	int i;
+
+	printk("[AD]: %u data_len %u\n", data->type, data->data_len);
+
+	switch (data->type) {
+	case BT_DATA_UUID16_SOME:
+	case BT_DATA_UUID16_ALL:
+		if (data->data_len % sizeof(uint16_t) != 0U) {
+			printk("AD malformed\n");
+			return true;
+		}
+
+		for (i = 0; i < data->data_len; i += sizeof(uint16_t)) {
+			struct bt_conn_le_create_param *create_param;
+			struct bt_le_conn_param *param;
+			const struct bt_uuid *uuid;
+			uint16_t u16;
+			int err;
+
+			memcpy(&u16, &data->data[i], sizeof(u16));
+			uuid = BT_UUID_DECLARE_16(sys_le16_to_cpu(u16));
+			if (bt_uuid_cmp(uuid, BT_UUID_HRS)) {
+				continue;
+			}
+
+			err = bt_le_scan_stop();
+			if (err) {
+				printk("Stop LE scan failed (err %d)\n", err);
+				continue;
+			}
+
+			printk("Creating connection with Coded PHY support\n");
+			param = BT_LE_CONN_PARAM_DEFAULT;
+			create_param = BT_CONN_LE_CREATE_CONN;
+			create_param->options |= BT_CONN_LE_OPT_CODED;
+			err = bt_conn_le_create(addr, create_param, param,
+						&conn_central);
+			if (err) {
+				printk("Create connection with Coded PHY support failed (err %d)\n",
+				       err);
```

Comment:
Surely you can fit this on a single line? "Coded PHY connection failed (err %d)", "Failed to create Coded PHY conn (err %d)", etc?